### ISSUE TITLE
High resolution image

### ISSUE THREAD

**Issue by Rikusen0335** [makegirlsmoe/makegirls.moe_web#3]
Please be able to export images with high resolution.

**Comment by AozakiOrenji** [comment]
with waifu2x? (x

**Comment by Aixile** [comment]
We described our attempts of generating high-resolution images in the paper. 

In our situation, the limitation of the resolution mainly comes from the training dataset. 
Personally, I don't think waifu2x gives a satisfying result on our generated images since the low-resolution images are not downscaled ones, it means that the low-resolution images lack necessary details for reconstructing high-frequency patterns in high-resolution images.

We also try to train our own SRGAN based super-resolution models, but not able to find a satisfying model yet. Therefore we leave the exploration of super-resolution models for future work.

**Comment by Skylion007** [comment]
@Aixile I might be able to help if you need a high res training set.

**Comment by YukiSakuma** [comment]
@Aixile did you try yet using a high res training data set before?

**Comment by Anonymous066** [comment]
I am tried from this command to proceed new high-res images.
```
cd /tmp
cp index.png o.png
waifu2x-converter-cpp -m scale --scale_ratio 1.15 -i o.png -o o2.png; waifu2x-converter-cpp --scale_ratio 1.3 --noise_level 3 -i o2.png -o o.png;
waifu2x-converter-cpp -m scale --scale_ratio 1.15 -i o.png -o o2.png; waifu2x-converter-cpp --scale_ratio 1.3 --noise_level 3 -i o2.png -o o.png;
waifu2x-converter-cpp -m scale --scale_ratio 1.15 -i o.png -o o2.png; waifu2x-converter-cpp --scale_ratio 1.3 --noise_level 3 -i o2.png -o o.png;
```
`index.png` is your target image which is anime face.
The result is bad, the image is lack of some detail. It must solved from source.
It means the "data training" must use high-res images.

@Aixile Yes, you are right.

**Comment by shaform** [comment]
@Aixile The paper says:
> Finally, from 42000 face images in total from the face detector, we manually
check all anime face images and remove about 4% false positive and undesired images.
> We train our GAN model using only images from games released after 2005 and with scaling all
training images to a resolution of 128*128 pixels. This gives 31255 training images in total.

Did you also scale up small images to 128x128 or just use images larger than 128x128?

From figure 14, it appears that more than half of the 42000 faces are smaller than 128x128, so if there were 31255 training images, it suggests that the small images were also used. Did this have any negative effects?

**Comment by Aixile** [comment]
@shaform I used images larger than 80*80 in the original  42000 faces. Personally, I think 1.5x upsampling is still affordable.

**Comment by shaform** [comment]
@Aixile Thanks! BTW, do you think it's possible to just extract faces from episodes of animes? Maybe we could get high volume of high resolution, high quality faces in this way.

**Comment by Aixile** [comment]
@shaform  Yes, but one fundamental problem for building large and high-quality anime face dataset is that lbpcascade_animeface, the only anime face detection library available online, uses old-fashioned methods and has poor precision/recall...

We are still improving our method and the dataset. Currently, we have achieved a much better result on CelebA dataset compared to the model we published. Hopefully to release the new model several months later.

**Comment by rtanen** [comment]
The website can produce 256x256 images now, which might close this issue.
